frontend/src/pages/MainPage.tsx

```diff
@@ -72,6 +57,16 @@ function MainPage() {
     setIsMapExpanded(prev => !prev);
   };
 
+  const hoverRestaurantCard = (targetId: number) => {
+    setIsHoveredList(prev =>
+      prev.map(item => (item.cardId === targetId ? { cardId: item.cardId, isHovered: true } : item)),
+    );
+  };
+
+  const unHoverRestaurantCard = () => {
+    setIsHoveredList(prev => prev.map(item => ({ cardId: item.cardId, isHovered: false })));
+  };
```

Comment:
위의 코멘트에 이어서... 그럼 이 함수들도 다음과 같이 표현할 수 있겠죠?

```suggestion
  const hoverRestaurantCard = (targetId: number) => setIsHoveredList(targetId);

  const unHoverRestaurantCard = () => setIsHoveredList(null);
```